src/senselab/audio/tasks/speaker_verification/speaker_verification.py

```diff
@@ -0,0 +1,81 @@
+"""Audio Processing and Speaker Verification Module.
+
+This module provides functions for resampling audio using an IIR filter and
+verifying if two audio samples or files are from the same speaker using a
+specified model.
+"""
+
+import typing as ty
+
+import torch
+from scipy import signal
+from speechbrain.augment.time_domain import Resample
+from speechbrain.inference.speaker import SpeakerRecognition
+
+from senselab.audio.data_structures.audio import Audio
+
+
+def _resample_iir(audio: Audio, lowcut: float, new_sample_rate: int, order: int = 4) -> Audio:
+    """Resamples audio using an IIR filter.
+
+    Args:
+        audio (Audio): The audio signal to resample.
+        lowcut (float): The low cut frequency for the IIR filter.
+        new_sample_rate (int): The new sample rate after resampling.
+        order (int, optional): The order of the IIR filter. Defaults to 4.
+
+    Returns:
+        Audio: The resampled audio signal.
+    """
+    sos = signal.butter(order, lowcut, btype="low", output="sos", fs=new_sample_rate)
+    filtered = torch.from_numpy(signal.sosfiltfilt(sos, audio.waveform.squeeze().numpy()).copy()).float()
+    resampler = Resample(orig_freq=audio.sampling_rate, new_freq=new_sample_rate)
+    return Audio(waveform=resampler(filtered.unsqueeze(0)).squeeze(0), sampling_rate=new_sample_rate)
+
+
+def verify_speaker(
+    audio1: Audio,
```

Comment:
all out tasks accept multiple audio clips as input. why don't you adapt this method so that it accepts something like List[tuple(Audio, Audio)]?

Reply:
Done.